### PR TITLE
Ensure metric names in docs match actual metric names

### PR DESCRIPTION
**Pull request by ulysseskan**
On https://goteleport.com/docs/setup/reference/metrics/ , we had a customer report that:

* the metric name in the docs ( `heartbeat_connections_missed_total`) doesn't match the actual metric name - `heartbeats_missed_total`.

This PR corrects this.  

Also, I curled the teleport metrics endpoint on a controller node and noticed a bunch of other metrics that are missing (or named differently) in the docs as well:

```
backend_requests Number of write requests to the backend [Also fixed in this PR]
bpf_lost_command_events Number of lost command events. [missing, do we want to place this in a new BPF section?]
bpf_lost_disk_events Number of lost disk events. [missing, do we want to place this in a new BPF section?]
bpf_lost_network_events Number of lost network events. [missing, do we want to place this in a new BPF section?]
firestore_events_backend_write_requests Number of write requests to firestore events [missing]
firestore_events_backend_write_seconds Latency for firestore events write operations [missing]
proxy_ssh_sessions_total Number of active sessions through this proxy [missing]
```

Feel free to let me know where these should be placed and I can add them.

### CODE REVIEW

**Comment by ptgott** [comment]
> bpf_lost_command_events Number of lost command events. [missing, do we want to place this in a new BPF section?]
> bpf_lost_disk_events Number of lost disk events. [missing, do we want to place this in a new BPF section?]
> bpf_lost_network_events Number of lost network events. [missing, do we want to place this in a new BPF section?]

I think a new BPF section would work. If this only supports SSH, though, we could also add it to the "Teleport Nodes" section.

> firestore_events_backend_write_requests Number of write requests to firestore events [missing]
> firestore_events_backend_write_seconds Latency for firestore events write operations [missing]

We can add these to the "Auth Service and backends" section

> proxy_ssh_sessions_total Number of active sessions through this proxy [missing]

We can add this to the Proxy section

**Comment by ulysseskan** [comment]
Enhanced Session Recording via BPF is currently SSH only, but there is [customer demand for it to be extended to support Kubernetes](https://github.com/gravitational/teleport/issues/3530) as well.  So I'll break it out into it's own section.

File: docs/pages/setup/reference/metrics.mdx [hunk 1]
```diff
@@ -99,9 +99,12 @@ Teleport Cloud does not expose monitoring endpoints for the Auth Service and Pro
 | `backend_batch_write_seconds` | histogram | cache | Latency for backend batch write operations. |
 | `backend_read_requests_total` | counter | cache | Number of read requests to the backend. |
 | `backend_read_seconds` | histogram | cache | Latency for read operations. |
-| `backend_write_requests_total` | counter | cache | Number of write requests to the backend. |
+| `backend_requests` | counter | cache | Number of write requests to the backend. |
 | `backend_write_seconds` | histogram | cache | Latency for backend write operations. |
 | `cluster_name_not_found_total` | counter | Teleport Auth | Number of times a cluster was not found. |
+| `dynamo_requests_total` | counter | dynamodb | Number of requests to the DynamoDB API. |
```
Comment (ptgott) [inline docs/pages/setup/reference/metrics.mdx:105]: I think we can capitalize the "dynamodb" category the same way we do the product itself.

File: docs/pages/setup/reference/metrics.mdx [hunk 2]
```diff
@@ -99,9 +99,12 @@ Teleport Cloud does not expose monitoring endpoints for the Auth Service and Pro
...
+| `dynamo_requests` | counter | dynamodb |  Number of failed requests to the DynamoDB API. |
+| `dynamo_requests_seconds` | histogram | dynamodb | Latency of DynamoDB api requests. |
```
Comment (ptgott) [inline docs/pages/setup/reference/metrics.mdx:107]:
```suggestion
| `dynamo_requests_seconds` | histogram | DynamoDB | Latency of DynamoDB API requests. |
```

**Comment by alexfornuto** [comment]
Resolving the conflicts was going to be a whole thing, so I recreated this PR as #16857

**Pull request closed without merging**